File: src/main/java/astnodes/parameters/OutParameter.java
package astnodes.parameters;
import astnodes.exp.var.Register;

public class OutParameter extends Parameter {

    public OutParameter(Register outputName, Register register) {
        super(outputName, register);
    }
    
    // TODO hack

    @Override
    public String toString() {
        return String.format("%s: bv%d, Gamma_%s: bool", getName(), getName().size().get(), getName());
    }
}

File: src/main/scala/astnodes/parameters/OutParameter.scala
package astnodes.parameters

import astnodes.exp.variable.Register

case class OutParameter(name: Register, register: Register) extends Parameter(name, register) {
  override def toString: String = {
    String.format("%s: bv%d, Gamma_%s: bool", name, name.size.get, name)
  }
}